internal/openstackclient/openstackclient.go

```diff
@@ -0,0 +1,184 @@
+package openstackclient
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/gophercloud/gophercloud/v2"
+	"github.com/gophercloud/gophercloud/v2/openstack"
+	"github.com/gophercloud/gophercloud/v2/openstack/compute/v2/servers"
+	"github.com/gophercloud/gophercloud/v2/openstack/config"
+	"github.com/gophercloud/gophercloud/v2/openstack/config/clouds"
+	"github.com/gophercloud/gophercloud/v2/openstack/image/v2/images"
+	"github.com/gophercloud/gophercloud/v2/openstack/utils"
+	"github.com/mitchellh/mapstructure"
+)
+
+type AuthConfig struct {
+	// AuthFromEnv specifies whether to use environment variables for auth
+	AuthFromEnv bool
+
+	// Cloud is the name of the cloud config from clouds.yaml to use
+	Cloud string
+
+	// CloudsConfig is the path to the clouds.yaml file
+	CloudsConfig string
+
+	// NovaMicroversion is the microversion of the OpenStack Nova client. Default 2.79 (which should be ok for Train+)
+	NovaMicroversion string
+}
+
+// Some good known properties useful for setting up ConnectInfo
+//
+// See also: https://docs.openstack.org/glance/latest/admin/useful-image-properties.html
+type ImageProperties struct {
+	// Architecture that must be supported by the hypervisor.
+	Architecture string `json:"architecture,omitempty" mapstructure:"architecture,omitempty"`
+
+	// OSType is the operating system installed on the image.
+	OSType string `json:"os_type,omitempty" mapstructure:"os_type,omitempty"`
+
+	// OSDistro is the common name of the operating system distribution in lowercase
+	OSDistro string `json:"os_distro,omitempty" mapstructure:"os_distro,omitempty"`
+
+	// OSVersion is the operating system version as specified by the distributor.
+	OSVersion string `json:"os_version,omitempty" mapstructure:"os_version,omitempty"`
+
+	// OSAdminUser is the default admin user name for the operating system
+	OSAdminUser string `json:"os_admin_user,omitempty" mapstructure:"os_admin_user,omitempty"`
+}
+
+type Client interface {
+	GetImageProperties(ctx context.Context, imageRef string) (*ImageProperties, error)
+	ShowServerConsoleOutput(ctx context.Context, serverId string) (string, error)
+	GetServer(ctx context.Context, serverId string) (*servers.Server, error)
+	ListServers(ctx context.Context) ([]servers.Server, error)
+	CreateServer(ctx context.Context, spec servers.CreateOptsBuilder, hintOpts servers.SchedulerHintOptsBuilder) (*servers.Server, error)
+	DeleteServer(ctx context.Context, serverId string) error
+}
+
+var _ Client = (*client)(nil)
```

Comment:
What for? If for compile time check if struct implements interface, New should do that already.

Vim-Go has integrated `golangci-lint`, which makes it even more verbose...

Reply:
Well :D Same as above, copied from Gitlab's plugin implementation. Removed it.